ecl/hql/hqldsparam.cpp

```diff
@@ -168,9 +168,10 @@ IIdAtom * HqlGram::fieldMapFrom(IHqlExpression* mapping, IIdAtom * id)
 // either a abstract dataset 
 bool HqlGram::requireLateBind(IHqlExpression* funcdef, const HqlExprArray& actuals)
 {
-    ForEachItemIn(idx, actuals)
+    //ForEachItemIn(idx, actuals)
+    for (auto iter : actuals)
     {
-        IHqlExpression *actual = &actuals.item(idx);
+        IHqlExpression *actual = static_cast<IHqlExpression *>(iter);
```

Comment:
Use of array iteraor.  Note iter is actually the nth value, so probably should be renamed in this case e.g. curActual

Reply:
If you have to add these casts, I'm not sure we are gaining a lot. If the casts can be avoided, it might be worthwhile - but IMHO only if we are not sacrificing efficiency. If we have to second-guess what is happening behind the scenes in case it's inefficient, then we'll end up not trusting the mechanism and not using it.